sklearn/manifold/locally_linear.py

```diff
@@ -43,9 +44,9 @@ def barycenter_weights(X, Z, reg=1e-3):
 
     n_samples, n_neighbors = X.shape[0], Z.shape[1]
     if X.dtype.kind == 'i':
-        X = X.astype(np.float)
+        X = astype(X, np.float, copy=False)
     if Z.dtype.kind == 'i':
-        Z = Z.astype(np.float)
+        Z = astype(Z, np.float, copy=False)
```

Comment:
We could also pass through other float types, though. ``dtype=[np.float64, np.float32]``?